### PR TITLE
Fix duplicate secretRef

### PR DESCRIPTION
**Pull request by kroussou**
While upgrading anchore-engine chart from 1.12.1 to 1.12.7 I've noticed that an extra secretRef entry was added everywhere where existing secret was referenced.
```
        envFrom:
        - secretRef:
            name: anchore-engine-secret
        - secretRef:
            name: anchore-engine-secret
        - configMapRef:
```

This is generated by the following lines in deployment templates when `.Values.anchoreGlobal.existingSecret` is set in values file:
```
        - secretRef:
            name: {{ default (include "anchore-engine.fullname" .) .Values.anchoreGlobal.existingSecret }}
        - secretRef:
            name: {{ default (print (include "anchore-engine.fullname" .) "-admin-pass") .Values.anchoreGlobal.existingSecret }}
```

To avoid duplication and produce cleaner output I suggest replacing these lines everywhere with
```
        {{- if .Values.anchoreGlobal.existingSecret }}
        - secretRef:
            name: {{ .Values.anchoreGlobal.existingSecret }}
        {{- else }}
        - secretRef:
            name: {{ include "anchore-engine.fullname" . }}
        - secretRef:
            name: {{ print (include "anchore-engine.fullname" .) "-admin-pass" }}
        {{- end }}
```

### CODE REVIEW

**Comment by Btodhunter** [comment]
@kroussou Thanks for the PR! This looks like a good fix to the chart, nice catch. I'll approve after doing some manual testing.

File: stable/anchore-engine/templates/enterprise_ui_deployment.yaml
Comment (Btodhunter) [inline stable/anchore-engine/templates/enterprise_ui_deployment.yaml:103]: This should use `.Values.anchoreEnterpriseUi.existingSecret`

File: stable/anchore-engine/templates/enterprise_ui_deployment.yaml
Comment (Btodhunter) [inline stable/anchore-engine/templates/enterprise_ui_deployment.yaml:106]: This should use `anchore-engine.enterprise-ui.fullname`

File: stable/anchore-engine/templates/enterprise_feeds_upgrade_job.yaml
Comment (Btodhunter) [inline stable/anchore-engine/templates/enterprise_feeds_upgrade_job.yaml:50]: This should use `.Values.anchoreEnterpriseFeeds.existingSecret`

File: stable/anchore-engine/templates/enterprise_feeds_upgrade_job.yaml
Comment (Btodhunter) [inline stable/anchore-engine/templates/enterprise_feeds_upgrade_job.yaml:53]: This should use `anchore-engine.enterprise-feeds.fullname`

File: stable/anchore-engine/templates/enterprise_feeds_deployment.yaml
Comment (Btodhunter) [inline stable/anchore-engine/templates/enterprise_feeds_deployment.yaml:102]: This should use `anchore-engine.enterprise-feeds.fullname`

File: stable/anchore-engine/templates/enterprise_feeds_deployment.yaml
Comment (Btodhunter) [inline stable/anchore-engine/templates/enterprise_feeds_deployment.yaml:99]: This should use `.Values.anchoreEnterpriseFeeds.existingSecret`

**Review by Btodhunter — CHANGES_REQUESTED** [review]

**Comment by Btodhunter** [comment]
@kroussou This looks great! I just noticed a few secret references that need to be updated before we can merge.

**Comment by kroussou** [comment]
@Btodhunter Mess with EE secret refs is cleaned thanks to your comments

**Review by Btodhunter — APPROVED** [review]
LGTM!

**Pull request merged**
